game/scripts/vscripts/TutorialGraph/Steps.ts

```diff
@@ -113,6 +113,91 @@ export const moveUnit = (unit: tg.StepArgument<CDOTA_BaseNPC>, moveLocation: tg.
         })
 }
 
+/**
+ * Creates a tutorial step that orders a unit to use an ability.
+ * @param caster Unit that will use the ability.
+ * @param target Target of the ability.
+ * @param abilityName Name of the ability.
+ * @param orderType Type of unit order used for casting the ability with ExecuteOrderFromTable.
+ */
+export const useAbility = (caster: tg.StepArgument<CDOTA_BaseNPC>, target: tg.StepArgument<CDOTA_BaseNPC> | tg.StepArgument<Vector>, abilityName: tg.StepArgument<string>, orderType: UnitOrder) => {
+    return tg.step((context, complete) => {
+        const actualCaster = tg.getArg(caster, context)
+        const ability = actualCaster.FindAbilityByName(tg.getArg(abilityName, context)) as CDOTABaseAbility
+
+        let order: ExecuteOrderOptions = {
+            UnitIndex: actualCaster.GetEntityIndex(),
+            OrderType: orderType,
+            AbilityIndex: ability.GetEntityIndex(),
+            Queue: true
+        };
+
+        if (typeof tg.getArg(target, context) === typeof CDOTA_BaseNPC) {
+            if (orderType === UnitOrder.CAST_TARGET)
+                order.TargetIndex = (tg.getArg(target, context) as CDOTA_BaseNPC).GetEntityIndex()
+            else
+                order.Position = (tg.getArg(target, context) as CDOTA_BaseNPC).GetAbsOrigin()
+        }
+        else {
+            order.Position = (tg.getArg(target, context) as Vector)
+        }
+
+        ExecuteOrderFromTable(order)
+
+        complete()
+    })
+}
+
+/**
+ * Creates a tutorial step that orders mirana to fire arrows at points on a line connecting two end points. Runs forever.
+ * @param miranaUnit Mirana unit that will shoot arrows.
+ * @param startPoint Starting point of the line.
+ * @param endPoint Ending point of the line.
+ * @param playerHero Optional parameter used to reset the player hero's position if hit by an arrow.
+ */
+export const fireArrowsInArea = (miranaUnit: tg.StepArgument<CDOTA_BaseNPC_Hero>, startPoint: tg.StepArgument<Vector>, endPoint: tg.StepArgument<Vector>, playerHero?: tg.StepArgument<CDOTA_BaseNPC_Hero>) => {
```

Comment:
Better to put this one into the section that uses it since it will only be used there and isn't general purpose.